demos/damped_pendulum/damped_pendulum.jl

```diff
@@ -45,27 +45,13 @@ strategy = GridTraining(0.1)
 discretization = PhysicsInformedNN(chain, strategy)
 
 # Define neural Lyapunov structure
-function _pos_def(state::AbstractVector, fixed_point)
-    θ, ω = state
-    θ_eq, ω_eq = fixed_point
-    log(1.0 + (sin(θ) - sin(θ_eq))^2 + (cos(θ) - cos(θ_eq))^2 + (ω - ω_eq)^2)
-end
-function _pos_def(state::AbstractMatrix, fixed_point)
-    θ = state[1,:]
-    s = sin.(θ)
-    c = cos.(θ)
-    ω = state[2,:]
-
-    θ_eq = fixed_point[1,:]
-    s_eq = sin.(θ_eq)
-    c_eq = cos.(θ_eq)
-    ω_eq = fixed_point[2,:]
-
-    return @. log(1.0 + (s - s_eq)^2 + (c - c_eq)^2 + (ω - ω_eq)^2)
-end
 structure = PositiveSemiDefiniteStructure(
         dim_output;
-        pos_def = _pos_def
+        pos_def = function (state, fixed_point)
+            θ, ω = state
+            θ_eq, ω_eq = fixed_point
+            log(1.0 + (sin(θ)-sin(θ_eq))^2 + (cos(θ)-cos(θ_eq))^2 + (ω-ω_eq)^2)
+        end
```

Comment:
I don't quite get this. What does it mean for pos_def to be a function?

Reply:
`pos_def` is the $p(x)$ in [github.com/SciML/NeuralLyapunov.jl/pull/5](https://github.com/SciML/NeuralLyapunov.jl/pull/5 ). So it's a positive (semi-)definite function that's used to ensure the Lyapunov function is 0 at 0. I'm not married to the naming of the variables or functions, and actually think some might need to be changed for clarity, but that wasn't my top concern yet.

Reply:
gotcha